share/ipld/get.go

```diff
@@ -245,39 +281,50 @@ func GetLeavesByNamespace(
 					singleErr.Do(func() {
 						retrievalErr = err
 					})
-					log.Errorw("getSharesByNamespace: could not retrieve node", "nID", nID, "pos", j.pos, "err", err)
+					log.Errorw("getLeavesWithProofsByNamespace: could not retrieve node", "nID", nID, "pos", j.sharePos, "err", err)
 					span.SetStatus(codes.Error, err.Error())
 					// we still need to update the bounds
-					bounds.update(int64(j.pos))
+					bounds.update(int64(j.sharePos))
 					return
 				}
 
 				links := nd.Links()
 				if len(links) == 0 {
 					// successfully fetched a leaf belonging to the namespace
 					span.SetStatus(codes.Ok, "")
-					leaves[j.pos] = nd
+					leaves[j.sharePos] = nd
 					// we found a leaf, so we update the bounds
 					// the update routine is repeated until the atomic swap is successful
-					bounds.update(int64(j.pos))
+					bounds.update(int64(j.sharePos))
 					return
 				}
 
 				// this node has links in the namespace, so keep walking
 				for i, lnk := range links {
 					newJob := &job{
 						id: lnk.Cid,
+						// sharePos represents potential share position in share slice
+						sharePos: j.sharePos*2 + i,
 						// position represents the index in a flattened binary tree,
 						// so we can return a slice of leaves in order
-						pos: j.pos*2 + i,
+						pos: j.pos*2 + i + 1,
```

Comment:
I don't understand why we are adding 1 now?

Reply:
Code here was working, but comment was incorrect. Old `j.pos*2 + i` is tracking share coordinates in shares array. `j.pos*2 + i + 1` operation maps tree into 1d struct without overwrites. We need second one here to collect proofs in tree, because second one will override values for node links (node children).

Example for old algo:
root ids: 0
root links: 2*0 (overwrite), 2*0+1 (1)
next links: 0*2 (overwrite), 2*0+1 (overwrite), 2*1+1 (3), 2*1+1 (4)

Reply:
Ohhhhhhhhhhh! Ok thanks for the explanation